cmd/utils/cmd.go

```diff
@@ -416,76 +423,99 @@ func ExportHistory(bc *core.BlockChain, dir string, first, last, step uint64) er
 	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
 		return fmt.Errorf("error creating output directory: %w", err)
 	}
+
+	var (
+		filename   func(string, int, common.Hash) string
+		newBuilder func(io.Writer) any
+		add        func(any, *types.Block, types.Receipts, *big.Int) error
+	)
+
+	if f == Era1 {
+		filename = era.Filename
+		newBuilder = func(w io.Writer) any { return era.NewBuilder(w) }
+		add = func(b any, blk *types.Block, rcpt types.Receipts, td *big.Int) error {
+			return b.(*era.Builder).Add(blk, rcpt, td)
+		}
+	} else {
+		filename = era2.Filename
+		newBuilder = func(w io.Writer) any { return era2.NewBuilder(w) }
+		add = func(b any, blk *types.Block, rcpt types.Receipts, td *big.Int) error {
+			return b.(*era2.Builder).Add(*blk.Header(), *blk.Body(), rcpt, td, nil)
+		}
+	}
```

Comment:
This is kind of impressive, but also is what an `Interface` is for :)

If you want different builders with the same methods (like `Add`) you can create the interface type and implement the method for both.